src/operators_lazytensor.jl

```diff
@@ -331,8 +331,13 @@ function _tp_matmul_mid!(result, a::AbstractMatrix, loc::Integer, b, α::Number,
         size(b, loc) == size(a, 2) && size(result, loc) == size(a, 1) || throw(DimensionMismatch("Dimensions of Eye matrix do not match subspace dimensions."))
         d = min(size(a)...)
 
-        rmul!(result, β)
-        @strided result_r[:, 1:d, :] .+= α .* br[:, 1:d, :]
+        if !iszero(β)
```

Comment:
@amilsted oh no, I completely missed that as well. Looks like we're missing a case in the unit tests too, right?

Reply:
It was in the previous code, I just missed its importance and dropped it while putting it in this PR. Tests did actually catch the problem, but it might be better to have more explicit tests.